frontend/src/core/api/entity.js

```diff
@@ -141,18 +141,22 @@ export default class EntityAPI extends APIBase {
 
         const results = await this.fetch('/other-locales/', 'GET', payload, headers);
 
-        if (!Array.isArray(results)) {
+        if (typeof results !== 'object' || Array.isArray(results) || results == null) {
             return [];
```

Comment:
We shouldn't return `[]`, because `LocalesState.translations` will not be an `Array` anymore, it will be an nobject. I suggest you return `null`.

Reply:
When I return `null` here I get an error that `null is incompatible with array type`

Reply:
Right, that's why you'll also need to update `actions` and the `reducer`: https://github.com/mozilla/pontoon/pull/1508#discussion_r353759554.